--- aribb24.orig/src/decoder.c	2021-04-14 20:11:29.268147787 +0900
+++ aribb24/src/decoder.c	2021-04-14 20:25:39.166092646 +0900
@@ -255,6 +255,7 @@ static int decoder_push( arib_decoder_t
         decoder->b_need_next_region = true;
     }
 
+#if 0
     /* Check for new paragraph/region */
     if( decoder->i_charleft >= decoder->i_right )
     {
@@ -262,6 +263,7 @@ static int decoder_push( arib_decoder_t
         decoder->i_charbottom += decoder->i_charheight;
         decoder->b_need_next_region = true;
     }
+#endif
 
     /* Ignore making new region */
     bool b_skip_making_new_region = false;
@@ -584,6 +586,7 @@ static int decoder_handle_esc( arib_deco
                 *handle = decoder_handle_katakana;
                 return 1;
             case 0x39:
+            case 0x3a:
             case 0x3b:
             case 0x42:
                 *handle = decoder_handle_kanji;
@@ -1139,9 +1142,10 @@ static int decoder_handle_time( arib_dec
                     return 1;
                 break;
             default:
-                if( i_mode == 1 && c >= 0x40 && c <= 0x7F )
+                if( i_mode == 1 && c >= 0x40 && c <= 0x7F ) {
                     decoder->i_control_time += c & 0x3f;
                     return 1;
+                }
                 return 0;
         }
         if( i_mode == 0 )
@@ -1206,9 +1210,9 @@ static int decoder_handle_c1( arib_decod
             decoder->b_need_next_region = true;
             return 1;
         case 0x89: //MSZ
-            decoder->i_fontwidth_cur = decoder->i_fontwidth / 2;
+            decoder->i_fontwidth_cur = decoder->i_fontwidth * 0.8;
             decoder->i_fontheight_cur = decoder->i_fontheight;
-            decoder->i_horint_cur = decoder->i_horint / 2;
+            decoder->i_horint_cur = decoder->i_horint * 0.8;
             decoder->i_verint_cur = decoder->i_verint;
             decoder->i_charwidth = decoder->i_fontwidth_cur + decoder->i_horint_cur;
             decoder->i_charheight = decoder->i_fontheight_cur + decoder->i_verint_cur;
